scripts/checks.sh

```diff
@@ -409,6 +408,35 @@ function check_disk_expansion()
 	fi
 }
 
+function check_image_corruption()
+{
+	local images
+	local corrupted=()
+	local -i corrupted_count=0
+	# TODO: this command is filtering out any images with a size <1Kb (for delta-based images)
+	images=$(${TIMEOUT_CMD} ${ENG} image ls --format "{{.ID}} {{.Size}}" | awk '/![1-9]*B/{next;} {print $1}' | sort -u)
+
+	if (( ${#images[@]} > 0 )); then
+		for i in ${images};
+		do
+			# TODO: the timeout here is probably too short to be effective in most cases, but let's gather
+			# data and change if necessary
+			if ! ${TIMEOUT_CMD} ${ENG} save "${i}" > /dev/null ; then
```

Comment:
is this check not also sensitive issues with disk IO?

Reply:
It is definitely sensitive. As a precaution I've timed out the engine call normally, but I suspect that might be not long enough (cause too many timeouts instead of real results). I thought the timeout data was collected and returned separate from the failure to save, but I shall fix that.